app/uk/gov/hmrc/transitmovementsvalidator/models/errors/TransitMovementError.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsvalidator.models.errors
+
+import cats.data.NonEmptyList
+import play.api.libs.functional.syntax.toFunctionalBuilderOps
+import play.api.libs.functional.syntax.unlift
+import play.api.libs.json.OWrites
+import play.api.libs.json.__
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.transitmovementsvalidator.models.formats.CommonFormats
+
+sealed abstract class TransitMovementError extends Product with Serializable {
+  def message: String
+  def code: String
+}
```

Comment:
This sort of name is a bit of a nonsense. It's an Error, more importantly it's in the Validation Microservice, and doesn't actually handle a Transit Movement. Maybe just BaseError?

Reply:
Sure. This was originally lifted from `transit-movements` and is now used in `common-transit-convention-traders`. We should choose a name that is far better for it, I agree, but if we do it here, we should do it there (and consider if we want to replicate this error building structure there too).

Reply:
Well, I guess that's a TD ticket for you to write ;)